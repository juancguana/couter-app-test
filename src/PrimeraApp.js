
// import React, { Fragment } from 'react';
import React from 'react';
import PropTypes from 'prop-types';

const PrimeraApp = ({ saludo, subtitulo }) => {

  return (
    <>
    <h1> { saludo } </h1>
      {/* <pre> { JSON.stringify( saludo, null, 3) } </pre> */}
  <p>{subtitulo}</p>
    </>
  );
}

PrimeraApp.propTypes = {
  saludo: PropTypes.string.isRequired
}

PrimeraApp.defaultProps = {
  subtitulo: 'Soy un subtitulo'
}

export default PrimeraApp;

