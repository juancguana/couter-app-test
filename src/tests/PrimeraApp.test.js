import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { render } from '@testing-library/react';
import PrimeraApp from '../PrimeraApp';

describe('Pruebas en <PrimeraApp />', () => {
  // test('Debe mostrar el mensaje "Hola, soy Juan"', () => {
  //   const saludo = 'Hola, soy Juan';
  //   const { getByText } = render(<PrimeraApp saludo={saludo} />);
  //   expect(getByText(saludo)).toBeInTheDocument();
  // });
  test('Debe mostar <PrimeraApp /> correctamente', () => {
    const saludo = 'Hola, soy Juank';
    const wrapper = shallow(<PrimeraApp saludo={saludo} />);

    expect(wrapper).toMatchSnapshot();
  });

  test('Debe entregar el subtitulo enviado por props', () => {
    const saludo = 'Hola soy Goku';
    const subtitulo = 'Hola soy un subtitulo';

    const wrapper = shallow(
      <PrimeraApp saludo={saludo} subtitulo={subtitulo} />
    );

    const textoParrafo = wrapper.find('p').text();
    expect(textoParrafo).toBe(subtitulo);
  });
});
