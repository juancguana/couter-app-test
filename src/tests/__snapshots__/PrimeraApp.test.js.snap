// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp /> Debe mostar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
     
    Hola, soy Juank
     !!
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
